sdk/monitor/monitor-query/src/models/publicMetricsModels.ts

```diff
@@ -115,6 +115,16 @@ export interface MetricsQueryResult {
   getMetricByName(metricName: string): Metric | undefined;
 }
 
+export function getMetricByName(this: MetricsQueryResult, metricName: string): Metric | undefined {
+  return this.metrics.find((it) => it.name === metricName);
+}
+
+export function createMetricsQueryResult({
+  ...params
```

Comment:
Do we need to splat / spread this out and then do the same in 125?

How about

```ts
export function createMetricsQueryResult(metricsQueryResultData: Omit<MetricsQueryResult, "getMetricByName">): MetricsQueryResult {
  return { ...metricsQueryResultData, getMetricByName };
}
```

Reply:
this looks cleaner